letsApply/Controllers/TabBarViewController/HomeScreenViewController/HomeViewController/HomeScreenViewController.swift

```diff
@@ -5,217 +5,174 @@
-            metricsCollectionView.heightAnchor.constraint(equalToConstant: 120)
-        ])
     }
     
-    private func setupViewModel() {
-        viewModel.onDataUpdated = { [weak self] in
-            DispatchQueue.main.async {
-                self?.updateUI()
-            }
-        }
+    private func setupNavigationBar() {
+        navigationItem.searchController = searchController
+        navigationItem.hidesSearchBarWhenScrolling = false
     }
     
-    private func fetchUserProfile() {
-        if let uid = Auth.auth().currentUser?.uid {
-            viewModel.fetchUserProfile(uid: uid)
-        }
-    }
+    // MARK: - Load Data
     
-    private func updateUI() {
-        if let profile = viewModel.userProfile {
-            greetingLabel.text = "Hey, \(profile.name) 👋"
-            
-            if let profileUrl = profile.profilePictureUrl {
-                ProfilePictureService.shared.fetchProfilePicture(urlString: profileUrl) { [weak self] image in
-                    self?.profileImageView.image = image
-                }
-            } else {
-                profileImageView.image = UIImage(systemName: "person.circle.fill")
+    private func loadUserProfile() {
+        homeViewModel.fetchUserProfile { [weak self] profile in
+            self?.greetingLabel.text = "Hello, \(profile.name)!"
+            self?.homeViewModel.loadProfileImage(urlString: profile.profilePictureUrl) { image in
+                self?.profileImageView.image = image
+            }
+            self?.homeViewModel.fetchRelevantJobs(for: profile) { jobs in
+                self?.jobs = jobs
+                self?.jobsCollectionView.reloadData()
             }
         }
-        metricsCollectionView.reloadData()
     }
 }
 
-// MARK: - UICollectionViewDataSource
-extension HomeScreenViewController: UICollectionViewDataSource, UICollectionViewDelegate {
+// MARK: - UICollectionViewDataSource, UICollectionViewDelegateFlowLayout
+
+extension HomeScreenViewController: UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+    
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return 3 // Number of metrics
+        if collectionView == categoriesCollectionView {
+            return homeViewModel.categories.count
+        } else {
+            return jobs.count
+        }
     }
     
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: MetricCardCell.reuseIdentifier, for: indexPath) as? MetricCardCell else {
-            return UICollectionViewCell()
+        if collectionView == categoriesCollectionView {
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: JobCategoryCell.identifier, for: indexPath) as! JobCategoryCell
```

Comment:
Don't forget to use a guard statement against the `as!` statement. Otherwise if your cell happens to not exist, the app will crash.